queries.sql

```diff
@@ -1,10 +1,65 @@
 /*Queries that provide answers to the questions from all projects.*/
 
-SELECT * from animals WHERE name LIKE '%mon';
-SELECT name FROM animals WHERE EXTRACT(year FROM date_of_birth) BETWEEN 2016 AND 2019;
-SELECT name FROM animals WHERE neutered AND escape_attempts < 3;
-SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
-SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
-SELECT * FROM animals WHERE neutered;
-SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+BEGIN;
+
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT my_savepoint;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO my_savepoint;
+
+UPDATE animals
+SET weight = weight * -1
+WHERE weight < 0;
+
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) FROM animals;
+
+SELECT name, neutered, escape_attempts
+FROM animals
+WHERE escape_attempts = (
+    SELECT MAX(escape_attempts)
+    FROM animals
```

Comment:
Kindly write the query for `What is the minimum and maximum weight of each type of animal?` as well.

- This query should utilize both the `MIN` and `MAX` SQL commands.